ql/src/java/org/apache/hadoop/hive/ql/ddl/table/partition/PartitionUtils.java

```diff
@@ -108,11 +122,17 @@ public static Partition getPartition(Hive db, Table table, Map<String, String> p
   public static List<Partition> getPartitions(Hive db, Table table, Map<String, String> partitionSpec,
       boolean throwException) throws SemanticException {
     List<Partition> partitions;
-    try {
-      partitions = partitionSpec == null ? db.getPartitions(table) : db.getPartitions(table, partitionSpec);
-    } catch (Exception e) {
-      throw new SemanticException(toMessage(ErrorMsg.INVALID_PARTITION, partitionSpec), e);
-    }
+      if (table.getStorageHandler() != null && table.getStorageHandler().alwaysUnpartitioned()) {
+        partitions = new ArrayList<>();
+        partitions.add(getPartition(db, table, partitionSpec, throwException));
```

Comment:
that is a diff API, it supposed to return all partitions matching spec, not just 1

Reply:
Yes, it is used in case of compactions.
It is called from AlterTableCompactOperation.
It is used to parallelize compaction of partitioned table - when partition spec is not provided it creates one compaction request per partition. For now it is not done for Iceberg, but will be useful in next PRs in which we will implement it for Iceberg too.